models/workday_history/workday__monthly_summary.sql

```diff
@@ -0,0 +1,112 @@
+        avg(annual_currency_summary_primary_compensation_basis) as avg_employee_primary_compensation,
+        avg(annual_currency_summary_total_base_pay) as avg_employee_base_pay,
+        avg(annual_currency_summary_total_salary_and_allowances) as avg_employee_salary_and_allowances,
+        avg(days_as_employee) as avg_days_as_employee
+    from months_employed
+    where date_month >= {{ dbt.date_trunc("month", "effective_date") }}
+        and (date_month <= {{ dbt.date_trunc("month", "wph_end_employment_date") }}
+            or wph_end_employment_date is null)
+    group by 1
+),
+
+monthly_active_worker_metrics as (
+    
+    select date_month,
+        count(distinct worker_id) as active_workers,
+        avg(annual_currency_summary_primary_compensation_basis) as avg_worker_primary_compensation,
+        avg(annual_currency_summary_total_base_pay) as avg_worker_base_pay,
+        avg(annual_currency_summary_total_salary_and_allowances) as avg_worker_salary_and_allowances,
+        avg(days_as_worker) as avg_days_as_worker
+    from months_employed
+    where (date_month >= {{ dbt.date_trunc("month", "effective_date") }}
+        and date_month <= {{ dbt.date_trunc("month", "wh_end_employment_date") }})
+            or wh_end_employment_date is null
+    group by 1
+),
+
+monthly_summary as (
+
+    select 
+        monthly_employee_metrics.date_month as metrics_month,
+        monthly_employee_metrics.new_employees,
+        monthly_employee_metrics.churned_employees,
+        monthly_employee_metrics.churned_voluntary_employees,
+        monthly_employee_metrics.churned_involuntary_employees,
+        monthly_employee_metrics.churned_workers,
+        monthly_active_employee_metrics.active_employees,
+        monthly_active_employee_metrics.active_male_employees,
+        monthly_active_employee_metrics.active_female_employees,
+        monthly_active_worker_metrics.active_workers,
+        monthly_active_employee_metrics.active_known_gender_employees,
+        monthly_active_employee_metrics.avg_employee_primary_compensation,
+        monthly_active_employee_metrics.avg_employee_base_pay,
+        monthly_active_employee_metrics.avg_employee_salary_and_allowances,
+        monthly_active_employee_metrics.avg_days_as_employee,
+        monthly_active_worker_metrics.avg_worker_primary_compensation,
+        monthly_active_worker_metrics.avg_worker_base_pay,
+        monthly_active_worker_metrics.avg_worker_salary_and_allowances,
+        monthly_active_worker_metrics.avg_days_as_worker
+    from monthly_employee_metrics
+    left join monthly_active_employee_metrics 
+        on monthly_employee_metrics.date_month = monthly_active_employee_metrics.date_month
+    left join monthly_active_worker_metrics
+        on monthly_employee_metrics.date_month = monthly_active_worker_metrics.date_month
+    order by monthly_employee_metrics.date_month
+)
+
+select *
+from monthly_summary
+order by metrics_month
```

Comment:
Good call. This was mostly for development/validation purposes (since otherwise everything was very jumbled), but isn't necessary in the final model.